base_layer/core/tests/median_timestamp.rs

```diff
@@ -1,138 +0,0 @@
-    assert_eq!(median_timestamp, (header1_timestamp + header2_timestamp) / 2);
-
-    let median_timestamp =
-        get_median_timestamp(get_header_timestamps(db, 2, timestamp_count)).expect("median returned an error");
-    assert_eq!(median_timestamp, header1_timestamp);
-
-    let median_timestamp =
-        get_median_timestamp(get_header_timestamps(db, 4, timestamp_count)).expect("median returned an error");
-    assert_eq!(median_timestamp, header2_timestamp);
-}
-
-#[test]
-fn test_median_timestamp_odd_order() {
-    let network = Network::LocalNet;
-    let consensus_manager = ConsensusManagerBuilder::new(network).build();
-    let timestamp_count = consensus_manager.consensus_constants(0).get_median_timestamp_count() as u64;
-    let store = create_store();
-    let pow_algos = vec![PowAlgorithm::Blake]; // GB default
-    create_test_pow_blockchain(&store, pow_algos, &consensus_manager);
-    let mut timestamps = vec![store.fetch_block(0).unwrap().block().header.timestamp.clone()];
-    let height = store.get_chain_metadata().unwrap().height_of_longest_chain();
-    let mut median_timestamp = get_median_timestamp(get_header_timestamps(
-        &*store.db_read_access().unwrap(),
-        height,
-        timestamp_count,
-    ))
-    .expect("median returned an error");
-    assert_eq!(median_timestamp, timestamps[0]);
-    let pow_algos = vec![PowAlgorithm::Blake];
-    // lets add 1
-    let tip = store.fetch_block(store.get_height().unwrap()).unwrap().block;
-    append_to_pow_blockchain(&store, tip, pow_algos.clone(), &consensus_manager);
-    timestamps.push(timestamps[0].increase(120));
-    let height = store.get_chain_metadata().unwrap().height_of_longest_chain();
-    median_timestamp = get_median_timestamp(get_header_timestamps(
-        &*store.db_read_access().unwrap(),
-        height,
-        timestamp_count,
-    ))
-    .expect("median returned an error");
-    assert_eq!(median_timestamp, (timestamps[0] + timestamps[1]) / 2);
-
-    // lets add 1 that's further back then
-    let append_height = store.get_height().unwrap();
-    let prev_block = store.fetch_block(append_height).unwrap().block().clone();
-    let new_block = chain_block(&prev_block, Vec::new(), &consensus_manager);
-    let mut new_block = store.calculate_mmr_roots(new_block).unwrap();
-    timestamps.push(timestamps[0].increase(60));
-    new_block.header.timestamp = timestamps[2];
-    new_block.header.pow.pow_algo = PowAlgorithm::Blake;
-    store.add_block(new_block.into()).unwrap();
-
-    timestamps.push(timestamps[2].increase(60));
-    let height = store.get_chain_metadata().unwrap().height_of_longest_chain();
-    median_timestamp = get_median_timestamp(get_header_timestamps(
-        &*store.db_read_access().unwrap(),
-        height,
-        timestamp_count,
-    ))
-    .expect("median returned an error");
-    // Median timestamp should be block 3 and not block 2
-    assert_eq!(median_timestamp, timestamps[2]);
-}
```

Comment:
Why remove all of these?
We are losing test coverage.

Reply:
That function does not exist anymore. I agree that we need to add tests to cover a lot of the backend, but unfortunately most of the tests create maintenance issues. I should be able to easily add tests for the refactor. Clean code is important. It makes testing easier, allows easier changes, easier onboarding for new devs. The benefits are manifold.